### PR TITLE
Updated plugin to use the nodejs csscomb implementation

### PR DESCRIPTION
**Pull request by faceleg**
Please review and let me know if any changes are required before merging

### CODE REVIEW

**Comment by istro** [comment]
:+1: 
Thank you!
For the time being I just cloned your repo, but this should definitely get addressed soon :-)

**Comment by istro** [comment]
Not sure how it used to work with old implementation - but on errors would it notify me?

An error (which I also get when csscomb is run in command line) is this:

```
Please check the validity of the block starting from line #25

23 |     margin-right: $margin-default;
25*|     // We need the extra specification to override bootstrap
26 |     > a {
27 |       padding: $margin-small $margin-tab;

Gonzales PE version: 3.0.0-28
Syntax: scss
CSScomb Core version: 3.0.0
```

Comment is actually fine I think, but in any case - CLI gives an error, shouldn't plugin also do something other than fail silently?

**Comment by faceleg** [comment]
You're right - I only tried it on vanilla CSS that passed linting (I use syntastic to lint css, I guess that runs before this plugin).

I'll look at adding error output when I get a chance

**Comment by istro** [comment]
Yeah, it's fair to assume people would set up lint rules to match their csscomb rules, so I guess it's not that big of a deal.

I actually just set up [scss-lint](https://github.com/brigade/scss-lint) (also uses syntastic) to lint my sass, it's AWESOME, hehe. Today's the cleanup day!

Probably not super high priority, so now that all the styles are neat and combed lint will prevent me from messing it up :+1:

**Comment by faceleg** [comment]
One would hope that if someone is using csscomb they also know about linting, yes!  But I still think it is a good idea to have the plugin at least output something when it barfs instead of doing it in silence.

**Comment by tonyganch** [comment]
Ahoj! Sorry for delay, I'll take a look at this pr tomorrow.

File: README.md [hunk 1]
```diff
@@ -29,6 +31,16 @@ Add this to .vimrc:
 Bundle 'git://github.com/miripiruni/CSScomb-for-Vim.git'
 ```
 
+### With NeoBundle
+Add this to .vimrc:
+```
+NeoBundle 'faceleg/vim-csscomb', {
```
Comment (tonyganch) [inline README.md]: Should not there be `csscomb/vim-csscomb`?

File: README.md [hunk 2]
```diff
@@ -29,6 +31,16 @@ Add this to .vimrc:
...
+        \ 'build': {
+        \     'unix': 'npm install -g csscomb',
+        \     'mac': 'npm install -g csscomb'
+        \ }}
```
Comment (tonyganch) [inline README.md]: I'm not familiar with NeoBundle. Could you please explain what's happening here?

File: README.md [hunk 3]
```diff
@@ -40,3 +52,12 @@ Vim command:
 ```
 :CSScomb
 ```
+
+## Suggested Configuration
+
+```VIML
+"mnemonic: 'beatify css'
```
Comment (tonyganch) [inline README.md]: It took me a while to figure out that this comment is actually meaningful.
Maybe you could write some sort of explanation for those people who don't get along with `.vimrc` files at all, so they can understand what this line does. Something like: "Map `bc` to run CSScomb. `bc` stands for `beautify css`". Or something neater :)

File: README.md [hunk 4]
```diff
@@ -40,3 +52,12 @@ Vim command:
...
+autocmd FileType css noremap <buffer> <leader>bc :CSScomb<CR>
```
Comment (tonyganch) [inline README.md]: Why `css` only?

Comment (faceleg) [inline README.md]: Oversight

File: README.md (hunk 3 above)
Comment (faceleg) [inline README.md]: Done

**Comment by faceleg** [comment]
I've addressed the issues you noted @tonyganch!

**Comment by faceleg** [comment]
@istro I've updated this PR to include error output in case of csscomb failure

**Comment by istro** [comment]
:+1:

**Comment by faceleg** [comment]
Note this PR should fix: #6, #7, #8

File: plugin/csscomb.vim [hunk 5]
```diff
@@ -7,10 +7,17 @@
...
-    let res = system("php ".fnameescape(g:CSScombPluginDir."/exec.php"), content)
-    let lines = split(res, "\n")
-    call setline(a:line1, lines)
+    let content = getline(a:line1, a:line2)
+
+    let tempFile = tempname() . '.' . &filetype
+    call writefile(content, tempFile)
+    let systemOutput = system('csscomb ' . shellescape(tempFile))
+    if len(systemOutput)
+        echoerr split(systemOutput, "\n")[1]
```
Comment (tonyganch) [inline plugin/csscomb.vim]: Great idea to show errors

File: README.md [hunk 6]
```diff
@@ -20,18 +22,28 @@ See details at [wiki](https://github.com/miripiruni/CSScomb/wiki/Requirements).
...
+```
+
+### With NeoBundle
+Add this to .vimrc:
+```
+NeoBundle 'csscomb/vim-csscomb', {
+        \ 'build': {
+        \     'unix': 'npm install -g csscomb',
+        \     'mac': 'npm install -g csscomb'
+        \ }}
```
Comment (tonyganch) [inline README.md]: As I understand from neobundle's code, `build` is a command run once repo is cloned.
If so, line 16 is not really a requirement.
Also, here i see a list of available platforms: https://github.com/Shougo/neobundle.vim/blob/fd15e35a7a7c4e84ae62981cfa529ae7cc382330/autoload/neobundle/installer.vim#L66
We can assume there are not so many vim users on windows, but linux is worth mentioning.

File: README.md [hunk 7]
```diff
@@ -40,3 +52,12 @@ Vim command:
...
 :CSScomb
 ```
+
+## Suggested Configuration
+
+```VIML
+" Map bc to run CSScomb. bc stands for beautify css
+autocmd FileType css noremap <buffer> <leader>bc :CSScomb<CR>
+" Automatically comb your CSS on save
+autocmd BufWritePre,FileWritePre *.css,*.scss,*.sass silent! :CSScomb<CR>
```
Comment (tonyganch) [inline README.md]: Less?

File: README.md (hunk 6 above)
Comment (tonyganch) [inline README.md]: I gave it some more thought. `build` is actually run every time plugin is updated. But we don't need to install `csscomb` every time.
Maybe it'll be enough for this plugin to be a wrapper for globally installed package, nothing more? This way user can easily control which version of the package he has installed.
I understand that those lines above are just example, they don't affect the code you wrote, but nonetheless, for those of us who just copy-paste everything we see, should we remove those `build` fields at all?

Comment (faceleg) [inline README.md]: Personally I have my systems setup to allow sudo-less global npm installs and like my vim plugins to take care of themselves, but I see your point. Removed the build config.

File: README.md (hunk 7 above)
Comment (faceleg) [inline README.md]: Added

File: plugin/csscomb.vim (hunk 5 above)
Comment (faceleg) [inline plugin/csscomb.vim]: Thanks to @istro

**Comment by tonyganch** [comment]
Thank you so much for this pr!

**Comment by faceleg** [comment]
No problem, thanks for merging! :dancers:

**Pull request merged**